### PR TITLE
bug(VENLY-3890) - add ImxTokenDiscoveryService

### PR DESCRIPTION
**Pull request by ZhamakochyanErik**

### CODE REVIEW

File: provider-imx/src/main/java/io/venly/provider/imx/NativeImxTokenDiscoveryService.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import io.venly.provider.imx.dto.ImxTokenInfoDto;
+import network.arkane.provider.chain.SecretType;
+import network.arkane.provider.client.BlockchainProviderGatewayClient;
+import network.arkane.provider.token.NativeTokenDiscoveryService;
+import network.arkane.provider.token.TokenInfo;
+
+import java.util.Optional;
+
+public class NativeImxTokenDiscoveryService implements NativeTokenDiscoveryService {
```
Comment (aleksanderndoci) [inline provider-imx/src/main/java/io/venly/provider/imx/NativeImxTokenDiscoveryService.java:11]: Name this ImxTokenDiscorveryService

File: provider-imx/src/main/java/io/venly/provider/imx/NativeImxTokenDiscoveryService.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    private final BlockchainProviderGatewayClient blockchainProviderGatewayClient;
+
+    public NativeImxTokenDiscoveryService(BlockchainProviderGatewayClient blockchainProviderGatewayClient) {
+        this.blockchainProviderGatewayClient = blockchainProviderGatewayClient;
+    }
+
+    @Override
+    public Optional<TokenInfo> getTokenInfo(String tokenAddress) {
+        ImxTokenInfoDto imxTokenInfoDto = blockchainProviderGatewayClient.get("/api/tokens/%s".formatted(tokenAddress), ImxTokenInfoDto.class);
```
Comment (aleksanderndoci) [inline provider-imx/src/main/java/io/venly/provider/imx/NativeImxTokenDiscoveryService.java:21]: change this to `blockchainProviderGatewayClient.get("/api/tokens/{tokenAddress}", ImxTokenInfoDto.class, tokenAddress)`

File: provider-imx/src/main/java/io/venly/provider/imx/ImxAutoConfiguration.java [hunk 3]
```diff
@@ -49,6 +50,12 @@ ImxTransactionInfoService imxTransactionInfoService(ImxTransactionGateway imxTra
         return new ImxTransactionInfoService(imxTransactionGateway);
     }
 
+    @Bean
+    @ConditionalOnBean(BlockchainProviderGatewayClient.class)
+    NativeImxTokenDiscoveryService nativeImxTokenDiscoveryService(BlockchainProviderGatewayClient blockchainProviderGatewayClient) {
```
Comment (aleksanderndoci) [inline provider-imx/src/main/java/io/venly/provider/imx/ImxAutoConfiguration.java:55]: This can be final

**Review by aleksanderndoci — CHANGES_REQUESTED** [review]

File: provider-imx/src/main/java/io/venly/provider/imx/dto/ImxTokenInfoDto.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class ImxTokenInfoDto {
+
+    private String decimals;
```
Comment (aleksanderndoci) [inline provider-imx/src/main/java/io/venly/provider/imx/dto/ImxTokenInfoDto.java:14]: Shouldn't this be integer/long ?

**Review by aleksanderndoci — CHANGES_REQUESTED** [review]

File: provider-imx/src/main/java/io/venly/provider/imx/dto/ImxTokenInfoDto.java (hunk 4 above)
Comment (ZhamakochyanErik) [reply provider-imx/src/main/java/io/venly/provider/imx/dto/ImxTokenInfoDto.java:14]: from ImxGateway getTokenInfo() method returns String 
I just cast in the blockchain-providers after getting from Imx

**Review by aleksanderndoci — APPROVED** [review]

**Pull request merged**